subsystems/elevatorsubsystem.py

```diff
@@ -0,0 +1,145 @@
+from enum import Enum, auto
+from math import pi
+from commands2 import Subsystem
+from ntcore import NetworkTableInstance
+
+from util.simtalon import Talon
+from util.convenientmath import clamp
+import constants
+
+
+class ElevatorSubsystem(Subsystem):
+    class ElevatorState(Enum):
+        L4Position = auto()
+        L3Position = auto()
+        L2Position = auto()
+        L1Position = auto()
+        AlgaeHigh = auto()
+        AlgaeLow = auto()
+        IntakePosition = auto()
+
+    def __init__(self) -> None:
+        Subsystem.__init__(self)
+        self.setName(__class__.__name__)  # basic subsystem boilerplate
+
+        self.elevatorMotor1 = Talon(
+            constants.kElevator1CANID,
+            constants.kElevator1Name,
+            constants.kElevator1PGain,
+            constants.kElevator1IGain,
+            constants.kElevator1DGain,
+            constants.kElevator1Inverted,
+            moMagicAccel=constants.kElevatorMaxAccel,
+            moMagicVel=constants.kElevatorMaxVel,
+        )
+
+        self.elevatorMotor2 = Talon(
+            constants.kElevator2CANID,
+            constants.kElevator2Name,
+            constants.kElevator2PGain,
+            constants.kElevator2IGain,
+            constants.kElevator2DGain,
+            constants.kElevator2Inverted,
+        )
+        self.elevatorMotor1.setNeutralMode(Talon.NeutralMode.Brake)
+        self.elevatorMotor2.setNeutralMode(Talon.NeutralMode.Brake)
+
+        self.elevatorMotor2.follow(self.elevatorMotor1, True)
+        self.targetPosition = 0
+
+        self.state = self.ElevatorState.L1Position
+
+        self.elevatorStatePublisher = (
+            NetworkTableInstance.getDefault()
+            .getStringTopic(constants.kElevatorStateKey)
```

Comment:
with new structured based publishing we should move away from the string topic for the state